sympy/core/add.py

```diff
@@ -1004,11 +1005,11 @@ def _eval_as_leading_term(self, x, cdir=0):
         except TypeError:
             return expr
 
-        new_expr=new_expr.together()
+        new_expr = new_expr.together().factor()
```

Comment:
When is `factor` needed?

Reply:
This is needed in - 
```py
def test_issue_17751():                                                         
    a, b, c, x = symbols('a b c x', positive=True)                              
    assert limit((a + 1)*x - sqrt((a + 1)**2*x**2 + b*x + c), x, oo) == -b/(2*a + 2)
```
At an intermediate step, we need `(a**2 + 2*a + 1)/_w**2` to be converted to `(a + 1)**2/_w**2`, or else the output remains - 
```py
      ⎛       ______________    ⎞
      ⎜      ╱  2               ⎟
∞⋅sign⎝a - ╲╱  a  + 2⋅a + 1  + 1⎠
```
(This cannot be further simplified in SymPy)

Reply:
It would be useful for maintenance to add a comment explaining the use of `factor`. What will happen if `together` is removed leaving `factor` alone?

Reply:
Yeah actually `together` can be removed. In some previous version, it was not possible to remove `together`, but it is now.